src/main/java/com/google/api/generator/spring/composer/SpringPropertiesClassComposer.java

```diff
@@ -256,7 +257,13 @@ private static List<MethodDefinition> createGetterSetters(
               String propertyName = Joiner.on("").join(methodAndPropertyName);
               getterAndSetter.add(
                   createGetterMethod(thisClassType, propertyName, propertyType, null));
-              getterAndSetter.add(createSetterMethod(thisClassType, propertyName, propertyType));
+              if (propertyType.equals(staticTypes.get("org.threeten.bp.Duration"))) {
+                // Use different setter logic for Duration handling conversion
+                getterAndSetter.add(
+                    createDurationSetterMethod(thisClassType, propertyName, propertyType));
+              } else {
+                getterAndSetter.add(createSetterMethod(thisClassType, propertyName, propertyType));
```

Comment:
Ah, that’s a good point! Currently the only call pattern is through `createGetterSetters()` since these are not exposed as public methods, but it will probably benefit future maintenance. I’ve moved the logic to all live inside `createSetterMethod()`.